packages/jest-core/src/getNoTestFoundFailed.ts

```diff
@@ -6,10 +6,20 @@
  */
 
 import chalk = require('chalk');
+import type {Config} from '@jest/types';
+import {isInteractive} from 'jest-util';
 
-export default function getNoTestFoundFailed(): string {
-  return (
-    chalk.bold('No failed test found.\n') +
-    chalk.dim('Press `f` to quit "only failed tests" mode.')
-  );
+export default function getNoTestFoundFailed(
+  globalConfig: Config.GlobalConfig,
+): string {
+  let msg = chalk.bold('No failed test found.');
+  if (isInteractive) {
+    msg += chalk.dim(
+      '\n' +
+        (globalConfig.watch
+          ? 'Press `f` to quit "only failed tests" mode.'
+          : 'Run Jest without `-f` or with `--all` to run all tests.'),
```

Comment:
```suggestion
          : 'Run Jest without `--only-failures` or with `--all` to run all tests.'),
```

I think that's cleaner than the alias